command/sub8_missions/sub8/sub_singleton.py

```diff
@@ -425,6 +426,97 @@ def go_to_sequence_quaternions(self, positions, orientations, speed=0.2):
                      orientations[i][2], orientations[i][3]))).go(speed)
 
 
+class SonarObjects(object):
+    def __init__(self, sub, pattern):
+        self.sub = sub
+        self.pattern = pattern
+        self._clear_pcl = self.sub.nh.get_service_client(
+            '/ogrid_pointcloud/clear_pcl', Trigger)
+
+        self._objects_service = self.sub.nh.get_service_client(
+            '/ogrid_pointcloud/get_objects', ObjectDBQuery)
+
+    @util.cancellableInlineCallbacks
+    def start(self, speed=0.5, clear=False):
+        if clear:
+            print 'SONAR_OBJECTS: clearing pointcloud'
+            yield self._clear_pcl(TriggerRequest())
+
+        print 'SONAR_OBJECTS: running pattern'
+        yield self._run_pattern(speed)
+
+        print 'SONAR_OBJECTS: requesting objects'
+        res = yield self._objects_service(ObjectDBQueryRequest())
+        defer.returnValue(res)
+
+    @util.cancellableInlineCallbacks
+    def start_until_found_x(self, speed=0.5, clear=False, object_count=0):
+        if clear:
+            print 'SONAR_OBJECTS: clearing pointcloud'
+            yield self._clear_pcl(TriggerRequest())
+        count = -1
+        while count < object_count:
+            for pose in self.pattern:
+                yield pose.go(speed=speed)
+                res = yield self._objects_service(ObjectDBQueryRequest())
+                count = len(res.objects)
+                if count >= object_count:
+                    defer.returnValue(res)
+        defer.returnValue(None)
+
+    @util.cancellableInlineCallbacks
+    def start_until_found_in_cone(self,
```

Comment:
Please tell someone how to use this. An example would be nice. That is pointed to in the comments of this specific file. Like one line above this preferably. Thank you.